src/server/db_setup.ts

```diff
@@ -103,7 +101,107 @@ export async function testQueryAll(): Promise<TestObject[]> {
+    if (Game_Results.length < 2) {
+        console.log("Too few players");
+        res.status(400).json({
+            status: 'error',
+            error: 'Must enter a minimum of 2 players',
+        });
+    }
+    else if (Game_Results.length > 6) {
+        // console.log("More than 6 players entered / this shouldn't be possible given there are only 6 places to input names");
+        res.status(400).json({
+            status: 'error',
+            error: "More than 6 players entered / this shouldn't be possible given there are only 6 places to input names",
+        });
+    }
+
+    //used for ordering players' ranking in DESC order
+    const playerPoints = new Map();
+    Game_Results.forEach((player : any) => {
+        const {playerName, victoryPoints} = player;
+        playerPoints.set(playerName, victoryPoints);
+    });
+
+    const playerPointsSorted = new Map([...playerPoints.entries()].sort((a, b) => b[1] - a[1]));
+    // console.log(playerPointsSorted);
+
+    //verify that a game_id of less than 10 characters is passed in because usually 8 or 9 characters (e.g. 20200911 - YYYYMMDD or 20200911a - YYYYMMDDa)
+    if (Game_Results.length >= 2 && Game_Results.length <= 6 && GameId.length < 10) {
+        Game_Results.forEach((result: any) => {
+            //gives access to variables from result
+            const {playerName, victoryPoints} = result; //equivalent to const playerData = result.playerData; and const victoryPoints = result.victoryPoints
+
+            //"Error [ERR_HTTP_HEADERS_SENT]: Cannot set headers after they are sent to the client" is still happening and not sure why
+            //However, the correct 400 status and error message shows up in the browser console for the below if statements
+            //server validation
+            if (playerName === null || playerName === "") {
+                // console.log("Invalid Player Name");
+                res.status(400).json({
+                    status: 'error',
+                    error: 'Invalid Player Name',
+                });
+            } 
+            if (victoryPoints === null || !Number.isInteger(victoryPoints)) {
+                // console.log("Invalid Victory Points");
+                res.status(400).json({
+                    status: 'error',
+                    error: 'Invalid Victory Points',
+                });
+            }
+            if (invalidChars.test(playerName)){ //check to see if there are any invalid characters in the string (see above for invalid chars)
+                // console.log("Invalid characters");
+                res.status(400).json({
+                    status: 'error',
+                    error: 'Invalid characters',
+                });
+            }
+
+            let index = 0;
+            //if the player's name and the points match, use that index for the player's ranking in the game (need both to handle ranking ties)
```

Comment:
Ties are resolved by the order which they are submit. e.g. `[matt: 1, bob: 1]` results in matt in first, bob in second.

Reply:
I think the way it is now should be good, but this would be another good opportunity to push this into a function that we unit/jest test with various combinations